### PR TITLE
fix: added support for onModalDismiss

### LINKED ISSUE #672: Support onModalDismiss for navigateModalScreen
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by sagardspeed2**
## Describe your changes
Added support for onModalDismiss in navigateModalScreen

## Issue ticket number and link

Closes #672 

## Checklist before requesting a review

- [ ] I have performed a self-review of my code
- [ ] I have added tests
- [ ] I have added a changeset `pnpm changeset add`
- [ ] I have added example usage in the kitchen sink app

### CODE REVIEW

File: apps/kitchen-sink/src/ensemble/screens/actions.yaml [hunk 1]
```diff
@@ -192,6 +192,10 @@ View:
                                 styles:
                                   fontSize: ${typography.fontSize['14']}
                                   fontWeight: ${typography.fontWeight.light}
+                      onDialogDismiss:
```
Comment (evshi) [inline apps/kitchen-sink/src/ensemble/screens/actions.yaml:195]: This should be onModalDismiss right?

**Review by evshi — COMMENTED** [review]
Since you're on this, can you also add support for `onDialogDismiss`?

File: apps/kitchen-sink/src/ensemble/screens/actions.yaml (hunk 1 above)
Comment (sagardspeed2) [reply apps/kitchen-sink/src/ensemble/screens/actions.yaml:195]: Yes, you are right, I renamed after test, and forgot to update the YAML

File: packages/runtime/src/runtime/hooks/useNavigateModal.tsx [hunk 2]
```diff
@@ -23,6 +29,16 @@ export const useNavigateModalScreen: EnsembleActionHook<
   const screenContext = useScreenContext();
   const customScope = useCustomScope();
   const storage = useEnsembleStorage();
+  const ensembleAction = useEnsembleAction(
+    isString(action) ? undefined : (action?.onModalDismiss as EnsembleAction),
```
Comment (evshi) [inline packages/runtime/src/runtime/hooks/useNavigateModal.tsx:33]: I don't think you need this - the `useEnsembleAction` hook will automatically return undefined.

**Review by evshi — APPROVED** [review]

**Pull request merged**